File: cortex-spring-boot-autoconfigure/src/main/java/ai/chatur/cortex/spring/ingest/IngestProperties.java
package ai.chatur.cortex.spring.ingest;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.bind.DefaultValue;
import org.springframework.core.io.Resource;

@ConfigurationProperties(prefix = "cortex.ingest")
public record IngestProperties(
    @DefaultValue(".cortex/db") String location,
    @DefaultValue("false") boolean persistent,
    @DefaultValue("classpath:shapes.ttl") Resource shapes) {}

File: src/main/kotlin/ai/chatur/cortex/ingest/IngestProperties.kt
package ai.chatur.cortex.ingest

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.bind.DefaultValue
import org.springframework.core.io.Resource

@ConfigurationProperties(value = "cortex.ingest")
data class IngestProperties(
    @DefaultValue("true") val enabled: Boolean,
    @DefaultValue("classpath:ontologies") val ontologiesPath: Resource,
    @DefaultValue("[]") val ontologies: List<String>,
    @DefaultValue("classpath:shapes") val shapesPath: Resource,
    @DefaultValue("[]") val shapes: List<String>,
)
